### PR TITLE
[TS] LPS-148048 Search result pagination should not contain a link to the same page to improve screen reader accessibility

### PR DESCRIPTION
**Pull request by jorgediaz-lr**
https://issues.liferay.com/browse/LPS-148048

_This issue has been reported by a blind association that is a Liferay customer, for more information see: https://issues.liferay.com/browse/LPS-148048?focusedCommentId=2626115&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-2626115_

The search result pagination has a different visual style to highlight which is the current page:
442718_pagination 1

But it also contains a link to itself, this link is processed by the screen readers, this is confusing for the final screen reader user as it is hard to notice that that link is a link to the current page.

To improve screen reader accessibility, this link should be removed.

After a quick research with several search engines:

- Google: https://www.google.com/search?q=search
- Bing: https://www.bing.com/search?q=search
- Yahoo: https://search.yahoo.com/search?p=search
- Amazon: https://www.amazon.com/s?k=search

All of them are not linking the current page, so it seems we should also remove the link.

I saw the same pattern in this example: https://weboverhauls.github.io/demos/pagination/

**Steps to reproduce**

1. Create 25 webcontent
2. Search "test"
3. Go to the pagination links and check the current page position:

   - **Expected behavior:** Current page doesn't have any link :heavy_check_mark:
   - **Wrong behavior:** Current page has a link :x:

### CODE REVIEW

**Comment by jorgediaz-lr** [comment]
ci:test:sf

**Comment by jorgediaz-lr** [comment]
ci:test:relevant

**Comment by jorgediaz-lr** [comment]
Hi @liferay-frontend 

This same issue is also reproduced using clay components.

Manually changing the html in the browser to disable the button of the current page fixes the issue, the screen reader doesn't read the current page.

I think changing this line https://github.com/liferay/clay/blob/8a854f2f867cb783f6311f881d326a5d8ed8af71/packages/clay-pagination/src/PaginationWithBasicItems.tsx#L139 will apply this solution to the clay code but I don't know how to update it and apply to the portal code.
Is this documented anywhere?

Thank you

**Comment by matuzalemsteles** [comment]
> Manually changing the html in the browser to disable the button of the current page fixes the issue, the screen reader doesn't read the current page.

Hey @jorgediaz-lr this wasn't very clear to me, are you saying that removing the `disabled` from the button or tag fixes the problem or adding the `disabled` fixes the problem?

It makes sense for the current/active element to remove the button or link tag to just plain text, it probably wouldn't be focusable either.

File: portal-web/docroot/html/taglib/ui/page_iterator/lexicon/start.jsp [hunk 1]
```diff
@@ -143,7 +152,14 @@ NumberFormat numberFormat = NumberFormat.getNumberInstance(locale);
 					%>
 
 						<li class="page-item <%= (i == cur) ? "active" : StringPool.BLANK %>">
-							<a class="page-link" href="<%= _getHREF(formName, namespace + curParam, i, jsCall, url, urlAnchor) %>" onclick="<%= forcePost ? _getOnClick(namespace, curParam, i) : "" %>"><span class="sr-only"><liferay-ui:message key="page" /></span><%= i %></a>
+							<c:if test="<%= i == cur %>">
+								<a aria-current="page" class="page-link">
```
Comment (diegonvs) [inline portal-web/docroot/html/taglib/ui/page_iterator/lexicon/start.jsp:156]: Very interesting!

When taking a look on https://www.aditus.io/aria/aria-current/, this a11y issue can be reproduced in breadcrumbs navigation and date pickers.

**Comment by jorgediaz-lr** [comment]
Hi @matuzalemsteles @diegonvs

For clay pagination components it is necessary to disable the button, this avoids the screen reader to read it.

With some help from @georgel-pop-lr I finally was able to change the clay code and test it, I have sent a PR here: https://github.com/liferay/clay/pull/4697

I am just disabling the button, perhaps it could be completely removed but my clay and react knowledge is very limited.

About leaving the link "a" tag without href, I decided to leave it because:

1. It seems is not read by the screen reader (I have tested it using Microsoft Narrator on Windows 10)
2. It avoids having to adjust the page styles.
3. Bing search uses this approach see https://www.bing.com/search?q=search they have a "a" without href
4. Same approach is used here: https://weboverhauls.github.io/demos/pagination/ where they explain it has more advantages than disadvantages.

Nevertheless, I am not an expert in accessibility, so perhaps I am wrong.

Let me know if you need more information.

**Comment by jorgediaz-lr** [comment]
Hi @matuzalemsteles @diegonvs 

Can we forward this PR? Is it necessary to do any change?

Regards,
Jorge

**Comment by diegonvs** [comment]
Yes @jorgediaz-lr, CI failures at https://github.com/liferay-frontend/liferay-portal/pull/1950#issuecomment-1049979883 seems unrelated. Trying to forward...

**Comment by diegonvs** [comment]
ci:forward

**Comment by jorgediaz-lr** [comment]
Thanks @diegonvs

**Comment by jorgediaz-lr** [comment]
Resent here https://github.com/liferay-frontend/liferay-portal/pull/1975 after applying BChan requested changes.

**Pull request closed without merging**